src/server/terrain/biomes.zig

```diff
@@ -462,17 +464,22 @@ pub const BlockStructure = struct { // MARK: BlockStructure
 		allocator.free(self.structure);
 	}
 
-	pub fn addSubTerranian(self: BlockStructure, chunk: *ServerChunk, startingDepth: i32, minDepth: i32, x: i32, y: i32, seed: *u64) i32 {
+	pub fn addSubTerranian(self: BlockStructure, chunk: *ServerChunk, startingDepth: i32, minDepth: i32, slope: i32, erosion: f32, x: i32, y: i32, seed: *u64) i32 {
 		var depth = startingDepth;
+		var timer = @max(@as(i32, @intFromFloat(@as(f32, @floatFromInt(slope))*erosion)) - 1, 0);
 		for(self.structure) |blockStack| {
 			const total = blockStack.min + main.random.nextIntBounded(u32, seed, @as(u32, 1) + blockStack.max - blockStack.min);
 			for(0..total) |_| {
-				if(chunk.liesInChunk(x, y, depth)) {
-					chunk.updateBlockInGeneration(x, y, depth, blockStack.block);
+				if(timer <= 0) {
```

Comment:
For simple expression an early continue is preferable, since then there is no gap between the if statement and the action:
```zig
if(timer > 0) {
    timer -= 1;
    continue:
}
// old code, no indentation change
```